jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/ServerWebSocketContainer.java

```diff
@@ -0,0 +1,278 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.websocket.server;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketBehavior;
+import org.eclipse.jetty.websocket.api.WebSocketContainer;
+import org.eclipse.jetty.websocket.api.WebSocketPolicy;
+import org.eclipse.jetty.websocket.api.WebSocketSessionListener;
+import org.eclipse.jetty.websocket.common.SessionTracker;
+import org.eclipse.jetty.websocket.core.WebSocketComponents;
+import org.eclipse.jetty.websocket.core.exception.WebSocketException;
+import org.eclipse.jetty.websocket.core.server.FrameHandlerFactory;
+import org.eclipse.jetty.websocket.core.server.WebSocketMappings;
+import org.eclipse.jetty.websocket.core.server.WebSocketNegotiator;
+import org.eclipse.jetty.websocket.server.internal.ServerFrameHandlerFactory;
+import org.eclipse.jetty.websocket.server.internal.ServerUpgradeRequestDelegate;
+import org.eclipse.jetty.websocket.server.internal.ServerUpgradeResponseDelegate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>A server-side WebSocket container that allows to {@link #addMapping(String, WebSocketCreator) map}
+ * URI paths to WebSocket endpoints and configure WebSocket parameters such as idle timeouts,
+ * max WebSocket message sizes, etc.</p>
+ */
+public class ServerWebSocketContainer extends ContainerLifeCycle implements WebSocketContainer, WebSocketPolicy
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ServerWebSocketContainer.class);
+
+    private final List<WebSocketSessionListener> listeners = new ArrayList<>();
+    private final SessionTracker sessionTracker = new SessionTracker();
+    private final Configuration configuration = new Configuration();
+    private final WebSocketMappings mappings;
+    private final FrameHandlerFactory factory;
+
+    ServerWebSocketContainer(WebSocketMappings mappings)
```

Comment:
Any reason for this to be private. Seems the only way for this to be created is from the `WebSocketUpgradeHandler`.

There is a use case for the `JettyWebSocketServerContainer` where no `WebSocketUpgradeFilter` or `WebSocketServlet` is ever created or any mappings added, and the upgrades can still be done manually through the container with the programmatic upgrade method.

Maybe this is something we want to support here as well?